src/getDataset.js

```diff
@@ -158,7 +166,12 @@ const getDataset = async (req, res) => {
         utils.verbose("Request is valid, but no dataset available. Returning 204.");
```

Comment:
> In some places, I inspect the error code itself (e.g. 404) and return a response with a status, and other times I throw/catch a custom Nextstrain NoContentError. I don't like how mixed my code currently is between these two approaches.

This seems like a good example of this. The code chooses to call either `requestCertainFileType` or `requestMainDataset`. The former throws custom exceptions, which we examine. The latter doesn't, so we examine the status code here. I believe you can modify the latter function to do something like
```js
Promise.reject(NoContentError);
```
but I haven't double checked. This would let you use consistent error handling within the `getDataset` handler.

Reply:
Thank you!

Reply:
Thanks!